### PR TITLE
Use weights parameter instead of deprecated pretrained for torchvision models

### PR DESCRIPTION
**Pull request by helioshe4**
## What changes are proposed in this pull request?

- Use weights parameter when loading torchvision models for people with torchvision >= 0.13.0
- Use pretrained=true parameter for users with torchvision < 0.13.0.  
- Backwards compatible for users with older versions of torchvision.

## How is this patch tested? If it is not, please explain why.

Downloaded various torchvision models using torchvision==0.15.2 and then using torchvision==0.12.0.
Tested model with no entrypoint_args.

### Is this a user-facing change that should be mentioned in the release notes?

-   [x] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

### CODE REVIEW

File: fiftyone/zoo/models/torch.py [hunk 1]
```diff
@@ -93,7 +93,12 @@ def _download_model(self, config):
 
     def _load_network(self, config):
         entrypoint = etau.get_function(config.entrypoint_fcn)
-        kwargs = config.entrypoint_args or {}
+
+        if version.parse(torchvision.__version__) < version.parse("0.13.0"):
+            kwargs = {"pretrained": True} if config.entrypoint_args else {}
```
Comment (brimoor) [inline fiftyone/zoo/models/torch.py:98]: For extra security, how about only pop-ing `weights'` and replacing it with `pretrained`?

**Review by brimoor — COMMENTED** [review]
@helioshe4 this is great, thanks! I left one comment for your consideration.

Also, could you update the destination branch for the PR to `release/v0.21.5` so we can include this patch in the next release?? 😄

File: fiftyone/zoo/models/torch.py (hunk 1 above)
Comment (helioshe4) [reply fiftyone/zoo/models/torch.py]: Sure, just pushed the new changes, hopefully I understood correctly.  I've also changed thee destination branch now.  Thanks for the feedback

Comment (helioshe4) [reply fiftyone/zoo/models/torch.py]: Created new PR off destination branch `release/v0.21.5` at #3348

**Pull request closed without merging**